From e1bed54db779d20b46d03fffd828fda987fea384 Mon Sep 17 00:00:00 2001
From: Pavel Picka <ppicka@redhat.com>
Date: Fri, 31 Jul 2020 16:29:01 +0200
Subject: [PATCH] PkgEnv copy depsolving bugfix

PkgEnv copy depsolving still used removed relations.
Now using group_ids and option_ids.

closes: #7248
https://pulp.plan.io/issues/7248

[nocoverage]

To test this new fixtures needed. Issue filled: https://pulp.plan.io/issues/7253
---
 CHANGES/7248.bugfix        | 1 +
 pulp_rpm/app/tasks/copy.py | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)
 create mode 100644 CHANGES/7248.bugfix

diff --git a/CHANGES/7248.bugfix b/CHANGES/7248.bugfix
new file mode 100644
index 000000000..38063a460
--- /dev/null
+++ b/CHANGES/7248.bugfix
@@ -0,0 +1 @@
+Fix copy with depsolving for packageenvironments.
\ No newline at end of file
diff --git a/pulp_rpm/app/tasks/copy.py b/pulp_rpm/app/tasks/copy.py
index 353a47789..37e72df54 100644
--- a/pulp_rpm/app/tasks/copy.py
+++ b/pulp_rpm/app/tasks/copy.py
@@ -91,7 +91,7 @@ def find_children_of_content(content, src_repo_version):
             packagegroups = packagegroups.union(category_package_groups)
 
     for packageenvironment in packageenvironments.iterator():
-        for env_package_group in packageenvironment.packagegroups:
+        for env_package_group in packageenvironment.group_ids:
             env_package_groups = PackageGroup.objects.filter(
                 name=env_package_group['name'], pk__in=src_repo_version.content
             )
@@ -99,7 +99,7 @@ def find_children_of_content(content, src_repo_version):
                 [envgroup.pk for envgroup in env_package_groups]
             )
             packagegroups = packagegroups.union(env_package_groups)
-        for optional_env_package_group in packageenvironment.optionalgroups:
+        for optional_env_package_group in packageenvironment.option_ids:
             opt_env_package_groups = PackageGroup.objects.filter(
                 name=optional_env_package_group['name'], pk__in=src_repo_version.content
             )
